docs/api/events.jmd

```diff
@@ -439,7 +444,13 @@ Useful for implementing an aggregate progress bar.
     ])
 }}
 
-{{ api_event("upload", "onUpload", "Called just before an item begins uploading to the server.",
+{{ api_event("upload", "onUpload",
+"""
+Called just before an item begins uploading to the server. In asynchronous work must be performed,
```

Comment:
@rnicholus We are still talking about returning Fine-uploader's own implementation of Promise and not native Promise?
Also in this line, should it be `In asynchronous` or `If asynchronous`?

Reply:
any "thenable" object is permitted

Reply:
though i dont want to encourage or promote Fine Uploader's promise impl at all. it's non-standard and just plain bad

Reply:
I'll keep the ability to return both ATM then.
Is the `In` supposed to be `If` there?

Reply:
I'd really prefer we don't even mention `qq.Promise` anymore. I expect/hope to rip it out entirely in 6.0.

Yes, it should be "if" instead of "in".

Reply:
Ok that's fine. I'll make changes accordingly then.